metron-deployment/packaging/ambari/metron-mpack/README.md

```diff
@@ -0,0 +1,127 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+This provides a Management Pack (MPack) extension for [Apache Ambari](https://ambari.apache.org/) that simplifies the provisioning, management and monitoring of Metron on clusters of any size.  
+
+This allows you to easily install Metron using a simple, guided process.  This also allows you to monitor cluster health and even secure your cluster with kerberos.
+
+### Prerequisites
+
+* Ambari 2.4.2+
+
+* Installable Metron packages (either RPMs or DEBs) located in a repository on each host at `/localrepo`.
+
+* A [Node.js](https://nodejs.org/en/download/package-manager/) repository installed on the host running the Management and Alarm UI.
+
+### Quick Start
+
+1. Build the Metron MPack. Execute the following command from the project's root directory.
+    ```
+    mvn clean package -Pmpack -DskipTests
+    ```
+
+1. This results in the Mpack being produced at the following location.
+    ```
+    metron-deployment/packaging/ambari/metron-mpack/target/metron_mpack-x.y.z.0.tar.gz
+    ```
+
+1. Copy the tarball to the host where Ambari Server is installed.
+
+1. Ensure that Ambari Server is stopped.
+
+1. Install the MPack.
+    ```
+    ambari-server install-mpack --mpack=metron_mpack-x.y.z.0.tar.gz --verbose
+    ```
+
+1. Install the Metron packages (RPMs or DEBs) in a local repository on each host where a Metron component is installed.  By default, the repository is expected to exist at `/localrepo`.
+
+    On hosts where only a Metron client is installed, the local repository must exist, but it does not need to contain Metron packages.  For example to create an empty repository for an RPM-based system, run the following commands.
+
+    ```
+    yum install createrepo
+    mkdir /localrepo
+    cd /localrepo
+    createrepo
+    ```
+
+1. Metron will now be available as an installable service within Ambari.  
+
+### Installation Notes
+
+The MPack will make all Metron services available in Ambari in the same manner as any other services in a stack.  These can be installed using Ambari's user interface using "Add Services" or during an initial cluster install.
+
+#### Co-Location
+
+1. The Parsers, Enrichment, Indexing, and Profiler masters should be colocated on a host with a Kafka Broker.  This is necessary so that the correct Kafka topics can be created.
+
+1. The Enrichment and Profiler masters should be colocated on a host with an HBase client.  This is necessary so that the Enrichment, Threat Intel, and Profile tables can be created.
+
+This colocation is currently not enforced by Ambari and should be managed by either a Service or Stack advisor as an enhancement.
```

Comment:
In the "Assign Masters" page, if the Kafka broker, Parsers host co-location or Alerts/Mgt UI co-location requirements are not met, then Ambari indicated this using an error balloon like:
![error](https://user-images.githubusercontent.com/20395490/34523079-ddc7106e-f0bb-11e7-9763-472a482d20a2.png)

If one of the above were not met and the user clicks "Next", then Ambari presents a validation issues dialog:
![validation-issues](https://user-images.githubusercontent.com/20395490/34523091-f0c13762-f0bb-11e7-902b-84d43722c3d8.png)

Now, the user can always hit "Continue Anyway" in the above dialog and move on with the wizard. This way, Ambari does not enforce the co-location.

The other client related requirements are indicated as a warning icon (since the client installation selection happens in the next step of the wizard).

(Thank you for reading through my long winding comment)

Reply:
Good find @anandsubbu .  So there are a few places that say basically "we should force collocation with the service advisor", but indeed that is already done.  I will remove that from the docs.

Reply:
Question 2: Do we even need to document the collocation requirements since they are enforced in the code?

Reply:
I checked the service_advisor.py script and all of these documented collocation requirements are enforced, except one.

I created https://issues.apache.org/jira/browse/METRON-1387 to track one missing collocation requirement.

Reply:
I removed the collocation requirements from the docs as these are enforced by the Mpack itself.  Let me know if anyone disagrees with this.